users/tests.py

```diff
@@ -362,6 +362,63 @@ def test_updating_user_recreates_instance_cache(self):
         self.assertEqual(len(cache.keys(cache_key_name)), 1)
         self.assertEqual(cache.get(cache_key_name)["first_name"], first_name)
 
+    def test_user_can_update_its_own_settings(self):
+        test_settings = json.dumps(
+            {
+                "player": {"configuration": {"skipEnabled": False}},
+                "app": {"appearance": {"darkMode": True}},
+            }
+        )
+        # update settings for user 1
+        response = self.client.put(
+            f"/api/v1/users/{self.user.id}/setting/",
+            json.loads(test_settings),
+            format="json",
+        )
+
+        # 200 OK returned as status
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        # The plio should contain the new updated settings object
+        self.assertEqual(
+            User.objects.filter(id=self.user.id).first().config["settings"],
+            json.loads(test_settings),
+        )
+
+    def test_only_superuser_can_update_other_user_settings(self):
+        test_settings = json.dumps(
+            {
+                "player": {"configuration": {"skipEnabled": False}},
+                "app": {"appearance": {"darkMode": True}},
+            }
+        )
+        # try to update settings for user_2
+        response = self.client.put(
+            f"/api/v1/users/{self.user_2.id}/setting/",
+            json.loads(test_settings),
+            format="json",
+        )
+
+        # Forbidden action
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        # make user 1 the superuser
```

Comment:
i don't think we need to write comments for every single action/line - this is fairly obvious - let's remove it?